crashreports/crashreporter/CrashReportApp.cpp

```diff
@@ -0,0 +1,464 @@
+        google_breakpad::MinidumpProcessor minidumpProcessor(nullptr, &resolver);
+        google_breakpad::MinidumpThreadList::set_max_threads(std::numeric_limits<uint32_t>::max());
+        google_breakpad::MinidumpMemoryList::set_max_regions(std::numeric_limits<uint32_t>::max());
+        
+        google_breakpad::ProcessState processState;
+        
+        if (minidumpProcessor.Process(&minidump, &processState) != google_breakpad::PROCESS_OK)
+        {
+            printf("Failed to process minidump");
+        }
+        else
+        {
+            google_breakpad::PrintProcessState(processState, true, &resolver);
+        }
+    }
+
+    wxString MakeDumpString(google_breakpad::Minidump& minidump, const wxString& temp)
+    {
+#if _WIN32
+        auto stream = _wfreopen(temp.wc_str(), L"w+", stdout);
+#else
+        auto stream = freopen(temp.utf8_str().data(), "w+", stdout);
+#endif
+        if (stream == NULL)
+            throw std::runtime_error("Failed to print minidump: cannot open temp file");
+        PrintMinidump(minidump);
+        fflush(stdout);
+
+        auto length = ftell(stream);
+        std::vector<char> bytes(length);
+        fseek(stream, 0, SEEK_SET);
+        fread(&bytes[0], 1, length, stream);
+        fclose(stream);
+        
+#if _WIN32
+        _wremove(temp.wc_str());
+#else
+        remove(temp.utf8_str().data());
+#endif
+
+        return wxString::From8BitData(&bytes[0], bytes.size());
+    }
+
+    wxString MakeHeaderString(google_breakpad::Minidump& minidump)
+    {
+        if (auto exception = minidump.GetException())
+        {
+            if (auto rawException = exception->exception())
+            {
+                // i18n-hint C++ programming assertion
+                return wxString::Format(_("Exception code 0x%x"), rawException->exception_record.exception_code);
+            }
+            else
+            {
+                // i18n-hint C++ programming assertion
+                return _("Unknown exception");
+            }
+        }
+        else if (auto assertion = minidump.GetAssertion())
+        {
+            auto expression = assertion->expression();
+            if (!expression.empty())
+            {
+                return expression;
+            }
+        }
+        return _("Unknown error");
+    }
+
+    void DoShowCrashReportFrame(const wxString& header, const wxString& dump, const std::function<bool(const wxString& comment)>& onSend)
+    {
+        static constexpr int MaxUserCommentLength = 2000;
+
+        auto frame = new wxFrame(
+            nullptr, 
+            wxID_ANY, 
+            _("Problem Report for Audacity"),
+            wxDefaultPosition, 
+            wxDefaultSize, 
+            wxDEFAULT_FRAME_STYLE & ~(wxRESIZE_BORDER | wxMAXIMIZE_BOX)//disable frame resize
+        );
+        frame->SetOwnBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_FRAMEBK));
+        
+        auto mainLayout = new wxBoxSizer(wxVERTICAL);
```

Comment:
2) sizer objects are not windows.  This should instead be:
`auto mainLayout = std::make_unique<wxBoxSizer>(wxVERTICAL);`

and ultimately you pass `mainLayout.release()` to SetSizer below.

This eliminates a suspicious naked new, guaranteeing cleanup.

(It would be better if the wxWidgets function took a std::unique_ptr argument, and you used `std::move(mainLayout)`.  But wxWidgets was developed for C++98 and just uses a plain pointer in the interface, while explaining in documentation that it does take over responsibility to delete it.  In C++11 and later, it is much better to let the types of arguments make this clear, and that is what we try to do in functions defined in Audacity.)